File: booking-backend/src/main/java/com/stenmartin/project/booking_backend/dto/model/ApiResponse.java
package com.stenmartin.project.booking_backend.dto.model;

import com.stenmartin.project.booking_backend.dto.entity.TireWorkshop;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ApiResponse<T> {
    private T result;
    private String statusCode;
    private String message;
    private boolean success;
    private TireWorkshop tireWorkshop;
}

File: booking-frontend/src/entity/ApiResponse.ts
import { TireWorkshop } from "./TireWorkshop";

export class ApiResponse<T> {
    result: T | null;
    statusCode: string;
    message: string;
    success: boolean;
    tireWorkshop: TireWorkshop;
  
    constructor(result: T | null = null, statusCode: string = '', message: string = '', success: boolean = false, tireWorkshop: TireWorkshop = new TireWorkshop()) {
      this.result = result;
      this.statusCode = statusCode;
      this.message = message;
      this.success = success;
      this.tireWorkshop = tireWorkshop;
    }
}
  